clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// VerifyConfig does a simple validation for config and all included structures.
+// Returns list of all not satisfied conditions.
+func VerifyConfig(c *api.Config) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.AutomanagedNamespaces < 0 {
+		allErrs = append(allErrs, field.Invalid(nil, "automanagedNamespaces", "must be non-negative"))
+	}
+	for i := range c.TuningSets {
+		allErrs = append(allErrs, verifyTuningSet(&c.TuningSets[i], field.NewPath("tuningSets").Index(i))...)
+	}
+	for i := range c.Steps {
+		allErrs = append(allErrs, verifyStep(&c.Steps[i], field.NewPath("steps").Index(i))...)
+	}
+	return allErrs
+}
+
+// verifyStep requires to either Phases or Measurements be non-empty,
+// but never both at the same time.
+func verifyStep(s *api.Step, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i := range s.Phases {
+		allErrs = append(allErrs, verifyPhase(&s.Phases[i], fldPath.Child("phases").Index(i))...)
+	}
+	for i := range s.Measurements {
+		allErrs = append(allErrs, verifyMeasurement(&s.Measurements[i], fldPath.Child("measurements").Index(i))...)
+	}
+	if (len(s.Phases) == 0) == (len(s.Measurements) == 0) {
+		allErrs = append(allErrs, field.Invalid(fldPath, "phases, measurements", "exactly one should non-empty"))
+	}
+	return allErrs
+}
+
+// verifyPhase requires ReplicasPerNamespace to have positive value.
+func verifyPhase(p *api.Phase, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if p.NamespaceRange != nil {
+		allErrs = append(allErrs, verifyNamespaceRange(p.NamespaceRange, fldPath.Child("namespaceRange"))...)
+	}
+	if p.ReplicasPerNamespace < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, "replicasPerNamespace", "must be non-negative"))
```

Comment:
This should be fldPath.Child("replicasPerNamespace").

There are also a number of other similar places - please fix.